pages/api/popular.ts

```diff
@@ -61,7 +67,23 @@ const PopularAssetsHandler: NextApiHandler = async (req, res) => {
   }
 
   try {
-    const dataProducts = await getPopularAssets([], `Bearer ${token.accessToken}`, limit)
+    const authString = `Bearer ${token.accessToken}`
+    const approvedStatusResp = await HttpClient.get<Collibra.Status>(`${config.COLLIBRA_BASE_URL}/statuses/name/Approved`, {
+      headers: { authorization: authString },
+    })
+
+    const dataProductTypeIdRes = await HttpClient.get<Collibra.PagedAssetTypeResponse>(`${config.COLLIBRA_BASE_URL}/assetTypes`, {
+      headers: { authorization: authString },
+      query: { name: "data product" },
+    })
+
+    const dataProducts = await getPopularAssets(
+      [],
+      authString,
+      approvedStatusResp.body.id,
+      dataProductTypeIdRes.body.results[0]?.id,
+      limit,
+    )
```

Comment:
i think a possible improvement we can do to this, is if `dataProductTypeIdRes.body.results[0].id` is `undefined` (meaning there are no "data product" types [which should never really be the case, but just to guard against it]), we can respond to the client with a 500 status code or something.

next.js' api handlers use a node server framework called express.js, so sending a 500 status code would look like this: `res.status(500).end()`